### PR TITLE
Project pull request

### PR DESCRIPTION
**Pull request by jdorety**
Not quite finished with MVP yet, will flag for review when I have gotten there.

### CODE REVIEW

**Comment by jdorety** [comment]
### Count Rules
- [x] balls and strikes reset to 0 when a player reaches 3 strikes or 4 balls.
- [x] balls and strikes reset to 0 when a hit is recorded.
- [x] a foul increases strikes up to 2. With no strikes, a foul makes it 1 strike. With 1 strike, a foul makes it 
  2 strikes. With two strikes a foul has no effect, count stays at 2 strikes.
### Display
- [x] display the count of balls and strikes for the at-bat.
- [x] should be updated when the user records activity on the Dashboard component.
### Dashboard
- [x] provide a button that the person in charge can press every time there is a strike, ball, foul or hit.
- [x] there is NO need to specify the type of hit (single, double, etc).
- [x] changes recorded on this component should update the information shown by the Display component.

File: src/components/Dashboard.test.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        fireEvent.click(ballButton);
+      }
+      getByText("2");
+      getByText("0");
+      cleanup();
+    });
+  });
+  describe("the hit button", () => {
+    it("should reset the count when pressed", () => {
+      const { getByText, getByTestId } = render(<Dashboard />);
```
Comment (johnoro) [inline src/components/Dashboard.test.js:57]: I think you could have this in a `beforeEach`, [docs here](https://jestjs.io/docs/en/setup-teardown#repeating-setup-for-many-tests):
```javascript
let getByText, getByTestId;
beforeEach(() => {
  { getByText, getByTestId } = render(<Dashboard />);
});
```
You could also take advantage of `afterEach` for your cleanup:
```javascript
afterEach(cleanup);
```

File: src/components/Dashboard.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    } else {
+      this.setState({ strikes: this.state.strikes + 1 });
+    }
+  };
+
+  ball = () => {
+    if (this.state.balls === 3) {
+      this.setState({ balls: 0, strikes: 0 });
+    } else {
+      this.setState({ balls: this.state.balls + 1 });
```
Comment (johnoro) [inline src/components/Dashboard.js:25]: Object destructuring would probably make this slightly prettier (the same goes for the `strike` method:
```javascript
ball = () => {
  const { balls } = this.state;
  if (balls === 3) {
    this.setState({ balls: 0, strikes: 0 });
  } else {
    balls++;
    this.setState({ balls });
  }
};
```
or, if you want to be clever while trading off readability:
```javascript
ball = () => {
  const { balls } = this.state;
  if (balls++ === 3) {
    this.setState({ balls: 0, strikes: 0 });
  } else {
    this.setState({ balls });
  }
};
```

File: src/components/Dashboard.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  hit = () => {
+    this.setState({ strikes: 0, balls: 0 });
+  };
+
+  foul = () => {
+    if (this.state.strikes <= 1) {
+      this.setState({ strikes: this.state.strikes + 1 });
+    } else return;
```
Comment (johnoro) [inline src/components/Dashboard.js:36]: This `else return` seems unnecessary as this function will return `undefined` regardless if the if condition is false.

**Review by johnoro — APPROVED** [review]
Looks great!

**Pull request merged**